### PR TITLE
Call cache diffing broken for scattered calls in carbonited workflows [BT-62]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Review by kshakir — APPROVED** [review]
I haven't traced why codecov is so unhappy with the diff, as upon first scan it does look like you've added regression tests.

Otherwise LGTM.

**Comment by mcovarr** [comment]
I'm suspicious the codecov issues may have been due to the sbt build failing, I've restarted that now 🤞

File: core/src/test/resources/hello_goodbye_papiv2.json [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    },
+    {
+      "cromwellId": "cromid-5b21691",
+      "cromwellVersion": "55-8d04562-SNAP",
+      "description": "PickedUp",
+      "timestamp": "2020-12-21T23:25:43.040Z"
+    }
+  ],
+  "workflowRoot": "gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/wf_hello/9b3f9d6d-67a9-405f-a55c-70550af02127/"
+}
```
Comment (grsterin) [inline core/src/test/resources/hello_goodbye_papiv2.json:242]: nitpicky: newline

File: core/src/test/resources/hello_goodbye_scattered_papiv2.json [hunk 2]
```diff
@@ -0,0 +1,766 @@
...
+    },
+    {
+      "cromwellId": "cromid-5b21691",
+      "cromwellVersion": "55-8d04562-SNAP",
+      "description": "Finished",
+      "timestamp": "2020-12-21T23:19:39.493Z"
+    }
+  ],
+  "workflowRoot": "gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/wf_hello/fd07f36a-3b17-4ae1-bf50-801ba6b773bd/"
+}
```
Comment (grsterin) [inline core/src/test/resources/hello_goodbye_scattered_papiv2.json:766]: newline

**Review by grsterin — APPROVED** [review]
LGTM and 2 nitpicky comments

**Review by kshakir — APPROVED** [review]

**Comment by kshakir** [comment]
To clear up my previous comment, it does indeed look like coverage was just an `sbt` test failure issue. Though I'm not clear on why the tests were failing. 🤷🤔:insert_flaky_test_emoji:

File: core/src/test/resources/hello_goodbye_papiv2.json (hunk 1 above)
Comment (mcovarr) [reply core/src/test/resources/hello_goodbye_papiv2.json:242]: this was just a jq massaging of raw metadata a la `curl http://my-cromwell/v1/workflow-id/metadata | jq . > workflow-id.json` that wasn't really intended to be read by humans...

**Pull request merged**